src/dune_file_watcher/dune_file_watcher.ml

```diff
@@ -403,12 +428,29 @@ let create_inotifylib_watcher ~sync_table ~(scheduler : Scheduler.t) =
                   None
               in
               match is_fs_sync_event_generated_by_dune with
-              | None -> process_inotify_event event
+              | None ->
+                let scan_dirs dirs =
+                  List.concat_map dirs ~f:(fun dir ->
+                      match Path.readdir_unsorted dir with
+                      | Error _ -> []
+                      | Ok contents ->
+                        List.map contents ~f:(fun fname ->
+                            let path = Path.relative dir fname in
+                            Inotify_lib.Event.Created (Path.to_string path)))
+                in
+                let rec loop acc = function
+                  | [] -> acc
+                  | event :: events ->
+                    let processed, to_scan =
+                      process_inotify_event ~inotify event
+                    in
+                    loop (acc @ processed) (events @ scan_dirs to_scan)
```

Comment:
The order of arguments in `@` seems weird, almost guaranteeing quadratic complexity if we assume that `acc` grows from a large number of small `processed` lists. Should this be swapped?

Reply:
(Same for `events`.)